src/main/java/org/protege/editor/owl/server/api/ServerLayer.java

```diff
@@ -148,8 +149,14 @@ public OWLOntology loadProjectSnapshot(ProjectId projectId) throws OWLOntologyCr
         }
     }
 
-    public String getSnapshotChecksum(ProjectId projectId) throws IOException {
+    public Optional<String> getSnapshotChecksum(ProjectId projectId) {
         Path path = Paths.get(getSnapShotFile(projectId).getAbsolutePath() + SNAPSHOT_CHECKSUM);
-        return new String(Files.readAllBytes(path), Charset.defaultCharset());
+        try {
+            return Optional.ofNullable(new String(Files.readAllBytes(path), Charset.defaultCharset()));
+        }
+        catch (IOException e) {
+            // snapshot is not present
+            return Optional.empty();
```

Comment:
It would be slightly preferably to return Optional.empty() in a file existence check rather than just converting any IO exception into an optional. Because this will also swallow things like permission errors for example.